ui/src/app/pages/admin/admin-workspace.tsx

```diff
@@ -0,0 +1,301 @@
+
+const LabelWithPrimaryColor = ({style = {}, children}) => {
+  return <label style={{color: colors.primary, ...style}}>
+    {...children}
+  </label>;
+};
+
+interface State {
+  googleProject: string;
+  workspace: Workspace;
+  collaborators: Array<UserRole>;
+  resources: AdminWorkspaceResources;
+  detailsOpen: boolean;
+}
+
+export class AdminWorkspace extends React.Component<{}, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      googleProject: '',
+      workspace: null,
+      collaborators: [],
+      resources: {
+        workspaceObjects: {
+          cohortCount: 0,
+          conceptSetCount: 0,
+          datasetCount: 0
+        },
+        cloudStorage: {
+          notebookFileCount: 0,
+          nonNotebookFileCount: 0,
+          storageBytesUsed: 0
+        },
+        clusters: []
+      },
+      detailsOpen: false
+    };
+  }
+
+  updateProject(newProject: string) {
+    this.setState({googleProject: newProject});
+  }
+
+  getFederatedWorkspaceInformation() {
+    workspaceAdminApi().getFederatedWorkspaceDetails(this.state.googleProject).then(
+      response => {
+        this.setState({
+          workspace: response.workspace,
+          collaborators: response.collaborators,
+          resources: response.resources
+        });
+      }
+    ).catch(error => {
+      console.error(error);
+    });
+  }
+
+  maybeGetFederatedWorkspaceInformation(event) {
+    if (event.key === 'Enter') {
+      return this.getFederatedWorkspaceInformation();
+    }
+  }
+
+  render() {
+    const {workspace, collaborators, resources, detailsOpen} = this.state;
+    return <div>
+      <h2>Manage Workspaces</h2>
+      <FlexRow style={{justifyContent: 'flex-start', alignItems: 'center'}}>
+        <label style={{marginRight: '1rem'}}>Google Project ID</label>
+        <TextInput
+            style={styles.fixedWidthWithMargin}
+            onChange={value => this.updateProject(value)}
+            onKeyDown={event => this.maybeGetFederatedWorkspaceInformation(event)}
+        />
+        <Button
+            style={{height: '1.5rem'}}
+            onClick={() => this.getFederatedWorkspaceInformation()}
+        >
+          Search
+        </Button>
+      </FlexRow>
+      {
+        workspace && <FlexColumn>
+          <h3>Workspace Admin Actions</h3>
+          <FlexRow style={{justifyContent: 'space-between'}}>
+            <Button>Shut down all VMs</Button>
+            <Button>Disable workspace</Button>
+            <Button>Disable all collaborators</Button>
+            <Button>Exclude from public directory</Button>
+            <Button>Log administrative comment</Button>
+            <Button>Publish workspace</Button>
+          </FlexRow>
+        </FlexColumn>
+      }
+      {workspace && (resources.workspaceObjects || resources.cloudStorage) &&
+        <FlexColumn style={{flex: '1 0 auto'}}>
+          <h3>Basic Information</h3>
+          <FlexRow>
+            <FlexColumnWithRightMargin>
```

Comment:
This is already pretty great, but how about a function like this: `workspaceInfoField(labelText, divContents)`